### PR TITLE
Introduce GenPlus.listOfSizedWithIndex

### PR DESCRIPTION
**Pull request by charleso**

### CODE REVIEW

File: src/main/scala/com/ambiata/disorder/GenPlus.scala [hunk 1]
```diff
@@ -1,9 +1,13 @@
 package com.ambiata.disorder
 
 import org.scalacheck._, Arbitrary._, Gen._
+import scala.collection.JavaConverters._
 
 object GenPlus {
 
+  def sizedInt(from: Int, to: Int): Gen[Int] =
```
Comment (charleso) [inline src/main/scala/com/ambiata/disorder/GenPlus.scala]: Not happy with this name

Comment (domdere) [inline src/main/scala/com/ambiata/disorder/GenPlus.scala]: `fromRange`, is that what it is?

Comment (domdere) [inline src/main/scala/com/ambiata/disorder/GenPlus.scala]: looks like `bounded` conveys the concept better than `sized`

Comment (markhibberd) [inline src/main/scala/com/ambiata/disorder/GenPlus.scala]: I would consider just call it `choose` for the same reason as `nonEmptyListOf` mirrors the Gen version.

Comment (markhibberd) [inline src/main/scala/com/ambiata/disorder/GenPlus.scala]: (don't actually care though, and what is there isn't that bad.

**Comment by markhibberd** [comment]
:goat:

File: src/main/scala/com/ambiata/disorder/GenPlus.scala [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
 import org.scalacheck._, Arbitrary._, Gen._
+import scala.collection.JavaConverters._
 
 object GenPlus {
 
+  def choose(from: Int, to: Int): Gen[Int] =
+    Gen.sized(n => Gen.choose(from, Math.max(from, Math.min(n, to))))
+
   // The Gen version of this function results in too many discarded tests
   def nonEmptyListOf[A](gen: => Gen[A]): Gen[List[A]] =
```
Comment (charleso) [inline src/main/scala/com/ambiata/disorder/GenPlus.scala]: I can't actually reproduce the normal version discarding values. I wonder whether it was something else. I'm tempted to remove this and `choose` but I guess there isn't much harm in leaving them for now.

**Pull request merged**